homeassistant/components/wirelesstag.py

```diff
@@ -0,0 +1,290 @@
+                "Error: failed to install local push notifications<br />",
+                title="Wireless Sensor Tag Setup Local Push Notifications",
+                notification_id="wirelesstag_failed_push_notification")
+        else:
+            _LOGGER.info("Installed push notifications for all tags.")
+
+    @property
+    def update_callback_url(self):
+        """Return url for local push notifications(update event)."""
+        return '{}/api/events/wirelesstag_update_tags'.format(
+            self.hass.config.api.base_url)
+
+    @property
+    def binary_event_callback_url(self):
+        """Return url for local push notifications(binary event)."""
+        return '{}/api/events/wirelesstag_binary_event'.format(
+            self.hass.config.api.base_url)
+
+    def handle_update_tags_event(self, event):
+        """Main entry to handle push event from wireless tag manager."""
+        _LOGGER.info("push notification for update arrived: %s", event)
+        for entity in self.entities:
+            if event.data.get('id') == entity.tag_id:
+                _LOGGER.info("Entity to update state: %s event data: %s",
+                             entity, event.data)
+                entity.update_tag_info(event)
+
+    def handle_binary_event(self, event):
+        """Handle push notifications for binary (on/off) events."""
+        _LOGGER.info("Push notification for binary event arrived: %s", event)
+        try:
+            tag_id = event.data.get('id')
+            event_type = event.data.get('type')
+            for sensor in self.binary_sensors:
+                if (tag_id == sensor.tag_id and
+                        event_type == sensor.device_class):
+                    sensor.on_binary_event(event)
+        except Exception as ex:  # pylint: disable=W0703
+            _LOGGER.error("Unable to handle binary event:\
+                          %s error: %s", str(event), str(ex))
+
+
+def setup(hass, config):
+    """Set up the Wireless Sensor Tag component."""
+    conf = config[DOMAIN]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+
+    try:
+        from wirelesstagpy import (WirelessTags, WirelessTagsException)
+        wirelesstags = WirelessTags(username=username, password=password)
+
+        platform = WirelessTagPlatform(hass, wirelesstags)
+        platform.load_tags()
+        hass.data[DOMAIN] = platform
+    except (ConnectTimeout, HTTPError, WirelessTagsException) as ex:
+        _LOGGER.error("Unable to connect to wirelesstag.net service: %s",
+                      str(ex))
+        hass.components.persistent_notification.create(
+            "Error: {}<br />"
+            "Please restart hass after fixing this."
+            "".format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+        return False
+
+    # listen to custom event
+    hass.bus.listen('wirelesstag_update_tags',
```

Comment:
We should use the `helpers.dispatcher` helper to communicate data between components and platforms.

Reply:
I really appreciate your continuous attention to quality of my code. I really tries to conform. To do so I need some `best practices` or example how it should be done. If you could point me to reference implementation or some documentation with example if usage - it would help. Specifically, I dont understand here what exactly should be changed. I use custom event (with changed name to per your request to contain platform name) to receive local push notification from wirelesstag manager (hw bridge). Since platform is my central point hosting all tags and keeping track who needs to get which event I do dispatch myself. Not really clear what should I delegate to `helpers.dispatcher`.

Reply:
Done.